test/integration/deep/opaqueports/opaque_ports_test.go

```diff
@@ -14,169 +15,277 @@ import (
+				scChecks: checks(
+					hashNoOutbondHTTPRequest,
+					// No authority here, because we are calling the pod directly.
+					hasOutboundTCPWithTLSAndNoAuthority,
+				),
+				appName:   opaqueSvcApp,
+				appChecks: checks(hasInboundTCPTraffic),
+			},
+			{
+				name:   "calling an unmeshed service",
+				scName: opaqueUnmeshedSvcSC,
+				scChecks: checks(
+					hashNoOutbondHTTPRequest,
+					// No authority here, because we are calling the pod directly.
+					hasOutboundTCPWithNoTLSAndNoAuthority,
+				),
+			},
+		}
+
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				err := TestHelper.RetryFor(30*time.Second, func() error {
+					if err := checkPodMetrics(ctx, opaquePortsNs, tc.scName, tc.scChecks); err != nil {
+						return fmt.Errorf("failed to check metrics for client pod:%w", err)
+					}
+					if tc.appName == "" {
+						return nil
+					}
+					if err := checkPodMetrics(ctx, opaquePortsNs, tc.appName, tc.appChecks); err != nil {
+						return fmt.Errorf("failed to check metrics for app pod:%w", err)
+					}
+					return nil
+				})
 				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
-				if !tcpMetricRE.MatchString(metrics) {
-					return fmt.Errorf("failed to find expected TCP metric when pod is opaque\n%s", metrics)
+					testutil.AnnotatedFatalf(t, "unexpected metric for client pod", "unexpected metric for client pod: %s", err)
 				}
-
-				return nil
 			})
+		}
+	})
+}
 
-			if err != nil {
-				testutil.AnnotatedFatalf(t, "unexpected metric output", "unexpected metric output: %s", err)
-			}
-		})
-
-		t.Run("expect inbound TCP connection metric with expected TLS identity for opaque service app", func(t *testing.T) {
-			// Check the slow cooker metrics
-			err := TestHelper.RetryFor(30*time.Second, func() error {
-				pods, err := TestHelper.GetPods(ctx, opaquePortsNs, map[string]string{"app": opaqueSvcSC})
-				if err != nil || len(pods) == 0 {
-					return fmt.Errorf("error getting pods\n%w", err)
-				}
-				metrics, err := getPodMetrics(pods[0], opaquePortsNs)
-				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
-				if httpRequestTotalMetricRE.MatchString(metrics) {
-					return fmt.Errorf("expected not to find HTTP outbound requests when service is opaque\n%s", metrics)
-				}
-				// Check the application metrics
-				pods, err = TestHelper.GetPods(ctx, opaquePortsNs, map[string]string{"app": opaqueSvcApp})
-				if err != nil {
-					return fmt.Errorf("error getting pods\n%w", err)
-				}
-				metrics, err = getPodMetrics(pods[0], opaquePortsNs)
-				if err != nil {
-					return fmt.Errorf("error getting metrics for pod\n%w", err)
-				}
-				if !tcpMetricRE.MatchString(metrics) {
-					return fmt.Errorf("failed to find expected TCP metric when pod is opaque\n%s", metrics)
-				}
+func waitForAppDeploymentReady(opaquePortsNs string) {
+	TestHelper.WaitUntilDeployReady(map[string]testutil.DeploySpec{
```

Comment:
Done in https://github.com/linkerd/linkerd2/commit/e69e1d47ac5d889fa92ddaf66ac1e7949bcbd957

I decided to keep `waitForAppDeploymentReady` because I like that it is short and the name explains what it does.